vmm/src/cpu.rs

```diff
@@ -803,7 +782,19 @@ impl CpuManager {
 
         // Only create vCPUs in excess of all the allocated vCPUs.
         for cpu_id in self.vcpus.len() as u8..desired_vcpus {
-            self.create_vcpu(cpu_id, entry_point, None)?;
+            self.create_vcpu(cpu_id)?;
+        }
+
+        Ok(())
+    }
+
+    pub fn configure_vcpus(&mut self, entry_point: Option<EntryPoint>) -> Result<()> {
+        // A VCPU is locked when running, in that case it should not be configured.
+        for vcpu in self.vcpus.clone() {
+            if vcpu.try_lock().is_err() {
```

Comment:
I don't particularly like this approach as i'm not 100% sure that that invariant holds all the time. Can we instead only configure vCPUs that we know we need to. Perhaps by storing some extra data in CpuManager.

Or alternatively i'm not 100% sure why this struct needs to be wrapped in `Arc<Mutex<..>>` most of the parameters are only assigned at initialisation time.

Reply:
Is there a reason that a vcpu has to be locked when it enters the `run()`: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/1cb1cff8823ebffe87ae5956ed38e74420ca7718/vmm/src/cpu.rs#L970

If not, may be we can try removing the `Mutex`?

Reply:
The correct solution is not to call `configure_vcpu()` twice. Why is this necessary?